tests/testthat/test-correlation-network.R

```diff
@@ -0,0 +1,172 @@
+    expect_equal(getLinkColorScheme(net), 'posneg')
+})
+
+test_that("we can build a Network from an edgeList data.frame", {
+    #w a weight column
+    edgeList <- data.frame(
+        source = c('a', 'b', 'c'),
+        target = c('b', 'c', 'a'),
+        correlationCoef = c(.8,.3,-.8),
+        pValue = c(.01,.001,.1)
+    )
+    net <- CorrelationNetwork(object = edgeList, linkColorScheme = 'none', pValueThreshold = NULL)
+    expect_equal(getNodeIds(net), c('a', 'b', 'c'))
+    expect_equal(getDegrees(net), c(2, 2, 2))
+    expect_equal(!is.null(getCoords(net)), TRUE)
+    expect_equal(getLinks(net)[[2]]@weight, .3)
+    expect_equal(getLinks(net)[[3]]@weight, .8)
+    expect_equal(getLinkColorScheme(net), 'none')
+    expect_equal(getDegrees(net), c(2, 2, 2))
+
+    #w a color scheme
+    edgeList <- data.frame(
+        source = c('a', 'b', 'c'),
+        target = c('b', 'c', 'a'),
+        correlationCoef = c(.8,.3,-.8),
+        pValue = c(.01,.001,.1)
+    )
+    net <- CorrelationNetwork(object = edgeList, linkColorScheme = 'posneg', pValueThreshold = NULL)
+    expect_equal(getNodeIds(net), c('a', 'b', 'c'))
+    expect_equal(getDegrees(net), c(2, 2, 2))
+    expect_equal(!is.null(getCoords(net)), TRUE)
+    expect_equal(getLinks(net)[[1]]@weight, .8)
+    expect_equal(getLinks(net)[[2]]@weight, .3)
+    expect_equal(getLinks(net)[[3]]@weight, .8)
+    expect_equal(getLinks(net)[[1]]@color, 1)
+    expect_equal(getLinks(net)[[2]]@color, 1)
+    expect_equal(getLinks(net)[[3]]@color, -1)
+    expect_equal(getLinkColorScheme(net), 'posneg')
+    expect_equal(getDegrees(net), c(2, 2, 2))
+
+    #w a pValueThreshold
+    edgeList <- data.frame(
+        source = c('a', 'b', 'c'),
+        target = c('b', 'c', 'a'),
+        correlationCoef = c(.8,.3,-.8),
+        pValue = c(.01,.001,.1)
+    )
+    net <- CorrelationNetwork(object = edgeList)
+    expect_equal(getNodeIds(net), c('a', 'b', 'c'))
+    expect_equal(getDegrees(net), c(2, 2, 2))
+    expect_equal(!is.null(getCoords(net)), TRUE)
+    expect_equal(getLinkColorScheme(net), 'posneg')
+    expect_equal(length(getLinks(net)), 2)
+    expect_equal(getLinks(net)[[1]]@weight, .8)
+    expect_equal(getLinks(net)[[2]]@weight, .3)
+
+    #w a correlationCoefThreshold
+    edgeList <- data.frame(
+        source = c('a', 'b', 'c'),
+        target = c('b', 'c', 'a'),
+        correlationCoef = c(.8,.3,-.8),
+        pValue = c(.01,.001,.1)
+    )
+    net <- CorrelationNetwork(object = edgeList, correlationCoefThreshold = .5, pValueThreshold = NULL)
+    expect_equal(getNodeIds(net), c('a', 'b', 'c'))
+    expect_equal(getDegrees(net), c(2, 2, 2))
+    expect_equal(!is.null(getCoords(net)), TRUE)
+    expect_equal(getLinkColorScheme(net), 'posneg')
+    expect_equal(length(getLinks(net)), 2)
+    expect_equal(getLinks(net)[[1]]@weight, .8)
+    expect_equal(getLinks(net)[[2]]@weight, .8) #second link is actually third link!!
```

Comment:
why?

Reply:
the correlation coef threshold.. thats the point of this test